hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -2522,6 +2522,20 @@ public FlushResultImpl flushcache(List<byte[]> families, boolean writeFlushReque
       LOG.debug(msg);
       return new FlushResultImpl(FlushResult.Result.CANNOT_FLUSH, msg, false);
     }
+    // cannot flush non-existing column families, and fail-fast
+    if (families != null) {
+      List<String> noSuchFamilies =
+        families.stream().filter(cf -> !getTableDescriptor().hasColumnFamily(cf))
+          .map(cf -> Bytes.toString(cf)).collect(Collectors.toList());
+      if (noSuchFamilies.size() > 0) {
+        String noSuchFamiliesMsg = String.format(
+          "There are non-existing families %s, we cannot flush the region %s, in table %s.",
+          noSuchFamilies, getRegionInfo().getRegionNameAsString(),
+          getTableDescriptor().getTableName().getNameAsString());
+        LOG.warn(noSuchFamiliesMsg);
```

Comment:
I don't think this should log at all. It's essentially an HTTP 400 kind of situation. Maybe a TRACE log, if at all. By all means, send the message back to the client, but i don't think the RS needs to emit this message.

Reply:
I also think the log here is unnecessary too. If you want to log this to trace bad requests, RegionServer will log some key information for FlushRegionRequest and ExecuteProceduresRequest. Is it possible to merge the information here into that log?